src/python/pants/engine/internals/parametrize.py

```diff
@@ -40,11 +41,26 @@ class Parametrize:
 
     args: tuple[str, ...]
     kwargs: FrozenDict[str, ImmutableValue]
+    is_group: bool
 
     def __init__(self, *args: str, **kwargs: Any) -> None:
+        object.__setattr__(self, "is_group", False)
         object.__setattr__(self, "args", args)
         object.__setattr__(self, "kwargs", FrozenDict.deep_freeze(kwargs))
 
+    def keys(self) -> tuple[str]:
+        return (f"parametrize_{hash(self.args)}:{id(self)}",)
```

Comment:
Wow. I'm shocked this works with `:`. That's pretty cool.

TIL that using **kwargs to receive unnamed keyword args on a function really does merely construct a dictionary, and the keys don't have to be valid arg names!

Reply:
Wait! That means this would be valid!

```suggestion
        return (f'parametrize("{self.group_name}", ...)',)
```

Or even

```suggestion
        return (f'parametrize("{self.args[0]}", ...)',)
```

Then we can still have the nice behavior of overriding groups from `__defaults__`.


```
>>> dict(**{'parametrize("a", ...)': 1}, **{'parametrize("a", ...)': 2})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: dict() got multiple values for keyword argument 'parametrize("a", ...)'
```

A bit hacky, but that makes for a nice error message, and there is no way this could interfere with actual module kwargs since they can't have a name with these special chars.

Reply:
I like the way you think here, and this is the easy way out. I already got better error messages, that includes the target address for the troublesome declaration. Regarding support for overriding any groups from `__defaults__`, I'd prefer to support that explicitly, which also leaves the door open to choose how the override happens (replace, merge, don't replace, error out, etc. could be configurable even..)